tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -751,3 +751,44 @@ function funcAcceptingIntArgs(int... scores) returns int {
     };
     return fn();
 }
+
+string moduleLevel = "module text";
+
+function errorConstructorReferenceTest1() returns error? {
+    string temp = "text";
+    var process = function() returns error? {
+        moduleLevel = "module text1";
+        return error("An Error", tempString = temp, moduleString = moduleLevel);
+    };
+    return process();
+}
+
+function errorConstructorReferenceTest2() returns function() returns error? {
+    string temp = "text";
+    var process = function() returns error? {
+        return error("An Error", tempString = temp);
+    };
+    return process;
+}
+
+function errorConstructorInClosureTest() {
+     var temp = errorConstructorReferenceTest1();
+     error err = <error>temp;
+     assert(err.detail().toString(), "{\"tempString\":\"text\",\"moduleString\":\"module text1\"}");
+
+     var tempfunc = errorConstructorReferenceTest2();
+     var tempErr = tempfunc();
+     err = <error>tempErr;
+     assert(err.detail().toString(), "{\"tempString\":\"text\"}");
```

Comment:
Shall we assert the individual fields instead of calling toString on the detail?